### PR TITLE
KeyBackup: Disable signalling remaining keys to upload count

### PR DESCRIPTION
**Pull request by BillCarsonFr**
Related to https://github.com/element-hq/element-web/issues/26783#issuecomment-1895318551

Keys count performance can be pretty bad on some configurations (big database in FF).
For now we disable signalling the remaining key count to upload on rust crypto. (used by react sdk `SecureBackupPanel.tsx` to provide some feedback).

**Note for reviewer:**
I considered caching the count by only doing it when the backup loop is started.
The problem is that at the end of each sync you might receive some new keys and that could start a backup loop, meaning that the send performances could be degraded after each sync.
So for now, it's just disabled until we improve the store performances (count shouldn't take 10s of seconds)

## Checklist

-   [ ] Tests written for new code (and old code if feasible)
-   [ ] Linter and other CI checks pass
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: src/rust-crypto/backup.ts [hunk 1]
```diff
@@ -342,20 +342,29 @@ export class RustBackupManager extends TypedEventEmitter<RustBackupCryptoEvents,
 
                 if (!request || this.stopped || !this.activeBackupVersion) {
                     logger.log(`Backup: Ending loop for version ${this.activeBackupVersion}.`);
+                    if (!request) {
+                        // nothing more to upload
+                        this.emit(CryptoEvent.KeyBackupSessionsRemaining, 0);
```
Comment (BillCarsonFr) [inline src/rust-crypto/backup.ts:347]: Some test relies on this event to be fired at the end of the backup loop.

**Review by t3chguy — APPROVED** [review]

File: src/rust-crypto/backup.ts (hunk 1 above)
Comment (richvdh) [reply src/rust-crypto/backup.ts:347]: this comment should probably be in the code, not the PR.

**Review by richvdh — APPROVED** [review]
lgtm but please move your comment into the code

**Comment by BillCarsonFr** [comment]
@t3chguy @richvdh Thanks for the feedback. I created a new PR that is keeping functionality and only computing room key count if needed and once per iteration
https://github.com/matrix-org/matrix-js-sdk/pull/4015

**Comment by BillCarsonFr** [comment]
Closing in favor of https://github.com/matrix-org/matrix-js-sdk/pull/4015

**Pull request closed without merging**